src/pages/VideoListingPage/VideoListingPage.jsx

```diff
@@ -1,31 +1,98 @@
 import VideoCards from '../../components/UI/Cards/VideoCards/VideoCards';
+import { useVideo } from '../../context/VideoContext';
+import {
+  HolderImg1,
+  HolderImg2,
+  HolderImg3,
+  HolderImg4,
+  HolderImg5,
+  HolderImg6,
+  HolderImg7,
+  HolderImg8,
+  HolderImg9
+} from "../../assets/Holders/holder";
 import "./VideoListingPage.css";
+import { category as CategoryList } from '../../data/category.data';
+import { URL } from 'uuid/dist/v35';
 
 function VideoListingPage() {
   const [videoLib, setVideoLib] = useState([]);
+  const { videoContextList, setVideoContextList } = useVideo();
+  const [category, setCategory] = useState();
+  const categories = ["music", "shoes", "React", "entertainment", "bollywood", "sports","trendings"];
+  //music shoes technology  entertainment bollywood sports trendings 
+  const location = useLocation();
+
+  useEffect(() => {
+    setTimeout(() => {
+      console.log("in use set", category, videoContextList)
+      if (category === undefined) {
+        if (!!location.search.split("=")[1]) {
+          console.log([...videoContextList?.filter((videoObj) => videoObj.snippet.tags === location.search.split("=")[1])]);
+          setVideoLib([...videoContextList?.filter((videoObj) => videoObj.snippet.tags === location.search.split("=")[1])]);
+        }
+      }
+      else if (category.length !== 0) {
+        setVideoLib([...videoContextList?.filter((videoObj) => videoObj.snippet.tags === category)])
+      } else { 
+        setVideoLib([...videoContextList])
+      }
+    }, 1000);
+   },[category])
   useEffect(() => {
     try {
       (async () => {
         var res = await axios.get("/api/videos");
         console.log(res);
         // console.log(res.data.products);
         setVideoLib(res.data.videos);
-
+        setVideoContextList(res.data.videos);
       })()
+
+
+      console.log(location.search.split("=")[1])
+      if (!!location.search.split("=")[1]) {
+        console.log("in use")
+        setTimeout(() => {
+          setCategory(location.search.split("=")[1]);
+        }, 1000);
+    }
     }
     catch (error) {
       console.log("Product list page error", error);
     }
   }, []);
+
+  const filterHandler = (item) => { 
+    setCategory((prev)=> prev === item ? "" : item);
+    // setVideoLib([...videoContextList?.filter((videoObj) =>   videoObj.snippet.tags === item)])
+  }
   return (
-    <div>
+    <div className='full-width'>
+      {/* <div className='categories-batch-container'>
+        {
+          categories.map((item) => (
+            <span className='categories-batch' onClick={()=>filterHandler(item)}> {item}</span>
+          ))
+        }
+      </div> */}
+      <ul className='icons-cateogry-layer'>
+      {CategoryList.map((item) => (
+        <li className='cateogory-container-batch-icons' onClick={()=>filterHandler(item.type)} key={item.name}>
+          <img src={item.icon} alt="category "/>
+          { item.name}
+        </li>
+      ))}
+      </ul>
```

Comment:
Suggestion: You could create a video component for video listing page to encapsulate logics in separate file and for ease of maintenance

Reply:
I get your  point but this is so small part to move into another page this just is to click on category and moving it into another file i guess there is not explicit data fetching for this in future when I fetch category from backend I will make separate page for this